csharp/README.md

```diff
@@ -1,206 +1,45 @@
-the test suite.
-
-> Note: Visual Studio 2022 version or higher is required for .NET Standard 2.0 builds.
-
-The .NET Standard build of iceboxnet and test applications target `net6.0` You can change
-the target framework by setting the `AppTargetFramework` property to a different
-
-Target Framework Moniker value, for example:
-
-```
-msbuild msbuild\ice.proj /p:"AppTargetFramework=net7.0"
+```csharp
+// Client application
+using var communicator = Ice.Util.initialize(ref args);
+var hello = HelloPrxHelper.uncheckedCast(
+    communicator.stringToProxy("hello:default -h localhost -p 10000"));
+hello.sayHello();
 ```
 
-This builds the test programs for `net7.0`. The target frameworks you specify
-must implement .NET Standard 2.0.
-
-#### Strong Name Signatures
-
-You can add Strong Naming signatures to the Ice assemblies by setting the
-following environment variables before building these assemblies:
-
- - `PUBLIC_KEYFILE` Identity public key used to delay sign the assembly
- - `KEYFILE` Identity full key pair used to sign the assembly
-
-If only `PUBLIC_KEYFILE` is set, the assemblies are delay-signed during the
-build and you must re-sign the assemblies later with the full identity key pair.
-
-If only `KEYFILE` is set, the assemblies are fully signed during the build using
-`KEYFILE`.
-
-If both `PUBLIC_KEYFILE` and `KEYFILE` are set, assemblies are delay-signed
-during the build using `PUBLIC_KEYFILE` and re-signed after the build using
-`KEYFILE`. This can be used for generating [Enhanced Strong Naming][6]
-signatures.
-
-*Strong Name Signatures can be generated only from Windows builds.*
-
-#### Authenticode Signatures
-
-You can sign the Ice binaries with Authenticode by setting the following
-environment variables before building these assemblies:
- - `SIGN_CERTIFICATE` to your Authenticode certificate
- - `SIGN_PASSWORD` to the certificate password
- - `SIGN_SHA1` the SHA1 has of the signing certificate
-
-*Authenticode can be generated only from Windows builds.*
-
-#### Building only the Test Suite
-
-You can build only the test suite with this command:
-```
-msbuild msbuild\ice.proj /p:ICE_BIN_DIST=all
-```
-
-This build retrieves and installs the `zeroc.ice.net` NuGet package if
-necessary.
-
-## Building on Linux or macOS
-
-### Linux and macOS Build Requirements
-
-You need the [.NET 6.0 SDK][4] or [.NET 7.0 SDK][5] to build Ice for .NET from source.
-
-### Compiling Ice for .NET on Linux or macOS
-
-Open a command prompt and change to the `csharp` directory:
-```
-cd csharp
-```
+```csharp
+// Server application
+using var communicator = Ice.Util.initialize(ref args);
 
-Then run:
+// Destroy the communicator on Ctrl+C or Ctrl+Break
+Console.CancelKeyPress += (sender, eventArgs) => communicator.destroy();
+var adapter = communicator.createObjectAdapterWithEndpoints("Hello", "default -h localhost -p 10000");
+adapter.add(new HelloI(), Ice.Util.stringToIdentity("hello"));
```

Comment:
We should include the code for HelloI, otherwise the sample is incomplete.

Maybe we should also rename HelloI to something else, since it's an unusual class name in C#.  For example: class Printer - it prints a message on the console each time it receives a `sayHello` .

Reply:
Updated to use Printer, like in the C# minimal demo from your PR